### PR TITLE
Pass on securityOpts from podInfraContainer to container added to pod.

### PR DESCRIPTION
**Pull request by rhatdan**
This is an incomplete fix, as it would be best for the libpod library to be in charge of coordinating the container's dependencies on the infra container. A TODO was left as such. UTS is a special case, because the docker library that namespace handling is based off of doesn't recognize a UTS based on another container as valid, despite the library being able to handle it correctly. Thus, it is left in the old way.

Signed-off-by: haircommander <pehunt@redhat.com>
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
@rhatdan Needs a rebase

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
@mheon @vrothberg @giuseppe @umohnani8 PTAL

I think we are close to passing the tests.

File: pkg/namespaces/namespaces.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+		if len(parts) != 2 || parts[1] == "" {
+			return false
+		}
+	default:
+		return false
+	}
+	return true
+}
+
+// IpcMode represents the container ipc stack.
```
Comment (TomSweeneyRedHat) [inline pkg/namespaces/namespaces.go]: A suggestion to mull over.  I'd make a leading comment like this for each section be large and in charge.
```
// ***************
// IpcMode represents the container ipc stack.
// ***************
```

Or some such, just to make it easier to scan through by eye.   (and no, I'm not completely intentionally trying to sneak in a COBOL flower box )

``

File: pkg/spec/createconfig.go [hunk 2]
```diff
@@ -417,12 +412,14 @@ func (c *CreateConfig) GetContainerCreateOptions(runtime *libpod.Runtime) ([]lib
 
 		options = append(options, libpod.WithIPCNSFrom(connectedCtr))
 	}
-	if IsPod(string(c.IpcMode)) {
-		options = append(options, libpod.WithIPCNSFromPod(pod))
-	}
```
Comment (TomSweeneyRedHat) [inline pkg/spec/createconfig.go]: So these three removed appends for options aren't needed?

**Comment by TomSweeneyRedHat** [comment]
One small question, otherwise LGTM.  Some of the tests are still giving you a run for your money @rhatdan

File: pkg/spec/createconfig.go (hunk 2 above)
Comment (rhatdan) [reply pkg/spec/createconfig.go]: Yes we are handling the pod configuration in podman not in libpod.

File: cmd/podman/create.go [hunk 3]
```diff
@@ -357,6 +357,33 @@ func configureEntrypoint(c *cli.Context, data *inspect.ImageData) []string {
...
+func configurePod(c *cli.Context, runtime *libpod.Runtime, namespaces map[string]string) (map[string]string, error) {
+	pod, err := runtime.LookupPod(c.String("pod"))
+	if err != nil {
+		return namespaces, err
+	}
+	podInfraID, err := pod.InfraContainerID()
+	if err != nil {
+		return namespaces, err
+	}
+	if (namespaces["pid"] == cc.POD) || (!c.IsSet("pid") && pod.SharesPID()) {
```
Comment (mheon) [inline cmd/podman/create.go]: Should we change this to cc.Pod instead of cc.POD to match our other constants?

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
Reminder
@mheon @umohnani8 @baude @TomSweeneyRedHat @vrothberg @giuseppe PTAL

**Comment by mheon** [comment]
LGTM
@rh-atomic-bot r+

**Comment by mheon** [comment]
@rh-atomic-bot retry

**Pull request closed without merging**